presto-main/src/main/java/io/prestosql/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -378,20 +379,18 @@ protected Type visitRow(Row node, StackableAstVisitorContext<Context> context)
         @Override
         protected Type visitCurrentTime(CurrentTime node, StackableAstVisitorContext<Context> context)
         {
-            if (node.getPrecision() != null &&
-                    node.getFunction() != LOCALTIMESTAMP &&
-                    node.getFunction() != CurrentTime.Function.TIMESTAMP &&
-                    node.getFunction() != LOCALTIME) {
-                throw semanticException(NOT_SUPPORTED, node, "non-default precision not yet supported");
-            }
-
             Type type;
             switch (node.getFunction()) {
                 case DATE:
                     type = DATE;
```

Comment:
It can't happen in practice -- the parser will never generate that. Really, what we should do is separate the various syntactic forms into different AST nodes instead of a union structure, like we have today.

Reply:
I'll add a check for now, anyway.